File: miscellaneous/activity-selection/max_activity.py
from collections import defaultdict


def getMaxActivities(start, finish):
    """
    
    start: list of activity start time
    finish: list of activity finish time
    """
    returnIndex = []
    i = 0
    n = len(start)

    returnIndex.append(i)

    for j in range(n):
        if start[j] >= finish[i]:
            returnIndex.append(j)
            i = j

    return returnIndex

File: miscellaneous/activity-selection/max_activity.js

function printMaxActivities(start, finish) {
    let n = start.length;
    let i = 0;

    let results = [];
    results.push(i);

    for (let j=0; j<n; j++){
        if (start[j] >= finish[i]) {
            results.push(j);
        }
    }
    return results;
}

let start  = [10, 12, 20]
let finish = [20, 25, 30]

res = printMaxActivities(start, finish);
console.log(res);